### ISSUE TITLE
[Internship Project]: Samuel Maina

### ISSUE THREAD

**Issue by GemmaTuron** [ersilia-os/ersilia#717]
### Summary

Hello,

This is a public issue for a virtual daily stand-up. We will use this to briefly share the tasks of the day and the challenges and advances made, so that we can ensure smooth support from the Ersilia mentors and alignment between daily tasks and overall internship goals.

### Scope

Initiative 🐋

### Objective(s)

Internship goals:

- Learn how to work in an Open Source Community
- Improve the infrastructure of the Ersilia Model Hub
- Improve the usability and platform compatibility of Ersilia Models, mainly via Docker Images
- Learn about AI/ML models for drug discovery
- Identify new relevant AI/ML models and datasets and add them to this [discussion](https://github.com/ersilia-os/ersilia/discussions/694)
- Incorporate new AI/ML models in the Hub
- Improve the documentation of the Ersilia Model Hub for users
- Improve the documentation of the Ersilia Model Hub for developers

### Team

| Role & Responsibility | Username(s) | 
|--|--|
| Intern | @samuelmaina |
| Mentor| @miquelduranfrigola  |
| Coordinator| @GemmaTuron  |

### Timeline

Before starting your work, line up a few tasks and short description. This should not take long. For example, it could be something like:
Wednesday 21st June
- [ ] Create Intern daily task tracker
- [ ] Update the Ersilia GitHub Project
- [ ] Make sure Git Actions start upon success of the other workflows
- [ ] Work on the documentation for model testing
- [ ] Attend the Drug Discovery intro by WCAIR

**Comment by GemmaTuron** [comment]
In this example, following the tasks of Wednesday 21st June:

- [ ]  Create Intern daily task tracker
- [ ]  Update the Ersilia GitHub Project
- [ ]  Make sure Git Actions start upon success of the other workflows
- [ ]  Work on the documentation for model testing
- [ ]  Attend the Drug Discovery intro by WCAIR

**Comment by samuelmaina** [comment]
- [x] Refactored and tested model  [eos9yy1](https://github.com/ersilia-os/eos9yy1)
  -  https://github.com/ersilia-os/eos9yy1/issues/12
  - https://github.com/ersilia-os/eos9yy1/issues/14
 

- [x] Refactored Model  [eos5505](https://github.com/ersilia-os/eos5505) -  https://github.com/ersilia-os/eos5505/issues/7 

- [x]  Solved dependency issues for 
   -  [eos9yy1](https://github.com/ersilia-os/eos9yy1) - https://github.com/ersilia-os/eos9yy1/pull/15
   - [eos7pw8](https://github.com/ersilia-os/eos7pw8) - https://github.com/ersilia-os/eos7pw8/pull/8
   - https://github.com/ersilia-os/eos2r5a  - https://github.com/ersilia-os/eos2r5a/pull/7
 

- [x]  Tested  Model [eos526j](https://github.com/ersilia-os/eos526j)  - https://github.com/ersilia-os/eos526j/issues/4
- [x]  Tested Model  [eos6ao8](https://github.com/ersilia-os/eos6ao8)    - https://github.com/ersilia-os/eos6ao8/issues/3

**Comment by GemmaTuron** [comment]
Hi @samuelmaina !

Thanks, please update your tasks for today, if you can include the [issue](https://github.com/ersilia-os/eos2r5a/issues/6) on model eos2r5a

**Comment by samuelmaina** [comment]
### 22nd June , 2023
- [x] Tested  [eos9tyg](https://github.com/ersilia-os/eos9tyg) -  https://github.com/ersilia-os/eos9tyg/issues/4 
- [x] Tested [eos2r5a](https://github.com/ersilia-os/eos2r5a) - https://github.com/ersilia-os/eos2r5a/issues/6
- [x]  Tested [eos9sa2](https://github.com/ersilia-os/eos9sa2)  on docker-   https://github.com/ersilia-os/eos9sa2/issues/3

- [x] Tested [eos6ao8](https://github.com/ersilia-os/eos6ao8) on Docker- https://github.com/ersilia-os/eos6ao8/issues/3#issuecomment-1602488844
- [x] Minor modification in  [eos5505](https://github.com/ersilia-os/eos5505)  - https://github.com/ersilia-os/eos5505/issues/7#issuecomment-1602385321

**Comment by samuelmaina** [comment]
Hi @GemmaTuron,
Today I tested  [eos9yy1](https://github.com/ersilia-os/eos9yy1) on docker (its working fine)  but  I found the issue is already closed. I had planned to test the model and refactor   
[eos97yu](https://github.com/ersilia-os/eos97yu) . Maybe you can another model for me to refactor.

**Comment by samuelmaina** [comment]
### 23rd June, 2023

- [x]  Tested  [eos9yy1](https://github.com/ersilia-os/eos9yy1) on Docker
- [x]  Refactored [eos97yu](https://github.com/ersilia-os/eos97yu) but I have this  error when getting the model from the `"flaml.joblib":` Will try to find the problem during the week
```
 File "/root/eos/repository/eos97yu/20230623144601_D219F9/eos97yu/artifacts/framework/code/predict.py", line 28, in predict_LogPe
    mdl = joblib.load(os.path.join(checkpoints_dir, "flaml.joblib"))
  File "/usr/samuelmayna/envs/eos97yu/lib/python3.7/site-packages/joblib/numpy_pickle.py", line 658, in load
    obj = _unpickle(fobj, filename, mmap_mode)
  File "/usr/samuelmayna/envs/eos97yu/lib/python3.7/site-packages/joblib/numpy_pickle.py", line 577, in _unpickle
    obj = unpickler.load()
  File "/usr/samuelmayna/envs/eos97yu/lib/python3.7/pickle.py", line 1088, in load
    dispatch[key[0]](self)
  File "/usr/samuelmayna/envs/eos97yu/lib/python3.7/pickle.py", line 1376, in load_global
    klass = self.find_class(module, name)
  File "/usr/samuelmayna/envs/eos97yu/lib/python3.7/pickle.py", line 1430, in find_class
    return getattr(sys.modules[module], name)
AttributeError: module 'flaml.model' has no attribute 'ExtraTreeEstimator'
  ```
- [x] Started refactoring [eos1amr](https://github.com/ersilia-os/eos1amr)  but  I have a hard time fetching the model.It is 1.35GB in size and will likely take time.  I will on it during the weekends .

**Comment by GemmaTuron** [comment]
Hi @samuelmaina !
Thanks for this update. It seems we are unable to unpickle the model appropriately, could you try to manually open this file? Run the joblib.load outside the ersilia environment and just try to run a prediction manually

**Comment by samuelmaina** [comment]
### 26th June, 2023
- [x] Tested  [eos22io](https://github.com/ersilia-os/eos22io) on CLI and Colab - https://github.com/ersilia-os/eos22io/issues/7?notification_referrer_id=NT_kwDOA1VX87M2ODcwODU4OTE0OjU1OTI0NzIz#issuecomment-1606736577
- [x] Solved and made a PR on [eos97yu](https://github.com/ersilia-os/eos97yu)  - https://github.com/ersilia-os/eos97yu/issues/5?notification_referrer_id=NT_kwDOA1VX87M2ODQzNjYzNjk5OjU1OTI0NzIz#issuecomment-1608671921

**Comment by samuelmaina** [comment]
27th June, 2023

- [x]  Tested  eos2b6f on cli and colab - https://github.com/ersilia-os/eos2b6f/issues/5#issuecomment-1609487532
- [x]  Tested  eos2b6f on Docker - https://github.com/ersilia-os/eos22io/issues/7#issuecomment-1608809950
- [x]  Anaylyzed  Analyzed  'Conda Environment Service' object has no attribute 'pid' error from  the https://github.com/ersilia-os/eos7pw8/actions/runs/5390669140/jobs/9786365326#step:7:189881 issue
- [x] Made progress  on the  eos1amr model - Was able to identify some cause of  invalid molecules being parsed as correct ones.

**Comment by GemmaTuron** [comment]
Hi @samuelmaina 
Please add to the list of today revising the eos7pw8 model and making sure the path to the model checkpoints is well specified

**Comment by samuelmaina** [comment]
### 28th June, 2023
- [x] Tested  eos2b6f on docker
- [x] Solved the path error and retested model [eos7pw8](https://github.com/ersilia-os/eos7pw8)  - https://github.com/ersilia-os/eos7pw8/pull/11#event-9666619279
- [x] Troubleshooting the empty ouput error for   https://github.com/ersilia-os/eos1amr
- [x]  Attended the WCAIR Drug Discovery short course 
- [x]  Held a meeting  with my mentor on the https://github.com/ersilia-os/eos1amr issue where he gave me some tips and also said he would look into the issue.

**Comment by GemmaTuron** [comment]
Hi @samuelmaina !

Great progress, please update the eos1amr model issue with the troubleshooting you are doing so I can provide further support.
Remember to pull the latest Ersilia code since we made updates yesterday!
I have also assigned ou a new model in case you get done with this one

**Comment by samuelmaina** [comment]
- [x] Tested eos46e on CLI and  Colab  https://github.com/ersilia-os/eos46ev/issues/5#issuecomment-1612396910
- [x] Tested eos6hy3 on CLI and Colab, no testing on docker due to some model errors    - https://github.com/ersilia-os/eos6hy3/issues/6#issuecomment-1613504394
- [x]  Refactored  and Dockerized eos6aun  - https://github.com/ersilia-os/eos6aun/pull/5
- [x] Continued working on https://github.com/ersilia-os/eos1amr to resolve the error

**Comment by samuelmaina** [comment]
##  29th  June , 2023
- [x] Tested eos46e on CLI and  Colab  https://github.com/ersilia-os/eos46ev/issues/5#issuecomment-1612396910
- [x] Tested eos6hy3 on CLI and Colab, no testing on docker due to some model errors    - https://github.com/ersilia-os/eos6hy3/issues/6#issuecomment-1613504394
- [x]  Refactored  and Dockerized eos6aun  - https://github.com/ersilia-os/eos6aun/pull/5
- [x] Continued working on https://github.com/ersilia-os/eos1amr to resolve the error

**Comment by samuelmaina** [comment]
### 30 June, 2023
- [x] Tested  eos6hy3  on Colab - https://github.com/ersilia-os/eos6hy3/issues/6#issuecomment-1614733831
- [x]  Continued working on https://github.com/ersilia-os/eos1amr  model
- [x]  Looked into the  ModelTester in the test cases issues - https://github.com/ersilia-os/ersilia/issues/729

**Comment by GemmaTuron** [comment]
Hi @samuelmaina 
Reminder that you need to update this daily so that mentors can see what work are you focusing on

**Comment by samuelmaina** [comment]
### 3rd July , 2023

- [x]  Refactoring eos6tg8,  Faced with numerous new stardard incorporation issues as the model is quite old -  https://github.com/ersilia-os/eos6tg8/issues/2
- [x]  Worked  from eos6m4j  Worked on removing the fingerpring parts but I some weird bugs -  https://github.com/ersilia-os/eos6m4j/issues/1  
- [x]   Worked  on eos59rr  model https://github.com/ersilia-os/eos59rr from issue https://github.com/ersilia-os/ersilia/issues/736#issuecomment-1618876507 . The model is  creating a docker image even though I am using --repo_path(it is saying python/conda installation is  not enough). The created image is not able to download packages specified  in a equirements.txt  in Dockerfile. 
- [x]  Continued to test if I can resolve the refactoring issue of eos1amr   -  https://github.com/ersilia-os/eos1amr/issues/1


Today, I  spent  my day  debugging  on the pending 4  models.  I have raised issues on where I am stuck but I will continue to do my best to find solution to the problems.

**Comment by samuelmaina** [comment]
### 4th July, 2023
- [x] Tested Model  eos6pbf on CLI  and Colab-  https://github.com/ersilia-os/eos6pbf/issues/2#issuecomment-1619541150
- [x]  Tested Model  eos4avb on CLI and Colab   - https://github.com/ersilia-os/eos4avb/issues/8#issuecomment-1619550307
-  Had an issue with internet connection hence I could not download the huge docker images for testing but tomorrow hope everything is alright.
- [x]  Continued to work on the previous issues as mentioned in day 3rd Jully, 2023 https://github.com/ersilia-os/ersilia/issues/717#issuecomment-1619713729 . I attended the Tuesday meeting and some solutions were mentioned that I could use to resolve the issues but its was quite late on my side hence will introduce them tomorrow..

**Comment by GemmaTuron** [comment]
Hi @samuelmaina 

What is the status of eos1amr? You also have work on models eos7pw8 (prioritaty to close this issue since it's been open for a few weeks) and if you finish everything you can move onto eos6tg8.
Coordinate with Emma on the eos7pw8 since he did a quite a bit of testing there

**Comment by samuelmaina** [comment]
I am still facing the same issue for eos1amr in codespaces, I have printed the model output , input and output file when fetching in codespaces  and everything seems okay but still no output.Since the original model is working, I will start refactoring afresh making sure everything is working at each step. 
I will first start on  eos7pw8 as required.
Thank you.

**Comment by samuelmaina** [comment]
### 5th July, 2023
- [x] Resolved the issues with model  eos7pw8 - https://github.com/ersilia-os/eos7pw8/issues/9#issuecomment-1621665160
- [x]  Refactored and made PR  eos1amr - https://github.com/ersilia-os/eos1amr/issues/1#issuecomment-1622266341

**Comment by GemmaTuron** [comment]
Hi @samuelmaina 

Let's focus today in understanding the changes in service.py in `eos1amr`, and good job on debugging `eos7pw8`, I have merged the PR!
I won't assign new models so you can focus on finishing the division of `eos6m4j` into two models

**Comment by samuelmaina** [comment]
### 7th July, 2023
- Played around with the artifact folder of `eos1amr` to see the different output so as to get why the service.py is not working,  still no success.
- [x] Tested `eos8h6g`  on CLI, Colab and Docker - https://github.com/ersilia-os/eos8h6g/issues/1?notification_referrer_id=NT_kwDOA1VX87M2OTI5MTA1NDczOjU1OTI0NzIz#issuecomment-1624313464
- [x] Tested  `eos4se9` on CLI, Colab and Docker  - https://github.com/ersilia-os/eos4se9/issues/4?notification_referrer_id=NT_kwDOA1VX87M2OTI5MDgxNTMyOjU1OTI0NzIz#issuecomment-1624394370

**Comment by GemmaTuron** [comment]
Hi @samuelmaina 

Please add the tests and what you have tried in your issue: https://github.com/ersilia-os/eos1amr/issues/1
You still have model eos6tg8 to refactor as well. I won't add new ones until these are complete

**Comment by samuelmaina** [comment]
###  7th July, 2023
- [x] Discuss and retest the eos1amr  - PR merged  https://github.com/ersilia-os/eos1amr/pull/3
- [x]  Discuss with my mentor on a side project to impact ersilia 
- [x]  Refactor eos59rr 
     - Was able to refactor but  there is a problem with shaping 2D output
     - Raised an issue with ersilia building a docker image to fetch the model when using --repo_path. https://github.com/ersilia-os/eos59rr/issues/1
- [x]  Resolved the test issues raised for eos7pw8 - https://github.com/ersilia-os/eos7pw8/issues/9

**Comment by samuelmaina** [comment]
### 9th July, 2023

- [ ] Finish the refactoring of eos59rr
- [ ] Finish the refactoring of  eos6m4j
- [ ] Test model eos8a4x https://github.com/ersilia-os/eos8a4x/issues/5#event-9751432202
- [ ] Retesting  eos8h6g to see if the null errors are recurring - https://github.com/ersilia-os/eos8h6g/issues/1#issuecomment-1624996499

**Comment by samuelmaina** [comment]
@GemmaTuron, today I don't have any issues except for retesting @eos8h6g. Can you merge the PRs https://github.com/ersilia-os/eos6m4j/pull/2 and https://github.com/ersilia-os/eos59rr/pull/2.
Also you can assign me a new task.

**Comment by samuelmaina** [comment]
### 10th July, 2023
- [ ]  Assigned model eos2mrz for testing but still on hold - https://github.com/ersilia-os/eos2mrz/issues/1#issuecomment-1628510915
- [x] Hold 1 hour a feature meeting with my mentor. DIscussed about how to automate the creation of pack.py , service.py and Dockerfile to ease incorporation and maintenance of models.
- [x]  Researched and begun mock implementation of the issue
- [x] Made few changes in model eos59rr -  https://github.com/ersilia-os/eos59rr/pull/2

**Comment by samuelmaina** [comment]
### 11th July, 2023
- [ ]  Create a detailed issue on  why the automation of creation of the service.py , pack.py and Dockerfile
- [ ]  Look for new model for incorporation
- [ ] Ask for more models for refactoring  and testing
- [ ] Look for areas where I can improve  the  code documentation

**Comment by GemmaTuron** [comment]
Hi @samuelmaina !

Can you please update me on the refactoring of eos6m4j? This model is quite important because it is used by ZairaChem. I've merged all PRs, so the changes are ready?
You still have model eos6tg8 as suggested in the tasks also

**Comment by samuelmaina** [comment]
Thanks @GemmaTuron , 
Thanks for reminding me,  my bad , I totally forgot about eos6tg8  model. I will work on it today.

**Comment by samuelmaina** [comment]
> Hi @samuelmaina !
> 
> Can you please update me on the refactoring of eos6m4j? This model is quite important because it is used by ZairaChem. I've merged all PRs, so the changes are ready? You still have model eos6tg8 as suggested in the tasks also

Yes the changes are ready. I am looking at the actions.

**Comment by samuelmaina** [comment]
[]  - Finish testing of the models eos4u6p, done docker testing yesterday -     https://github.com/ersilia-os/eos4u6p/issues/3
[] - Finish testing of eos2mrz ,tested on docker yesterday-  https://github.com/ersilia-os/eos2mrz/issues/1#issuecomment-1631238267
[] - Finish refactoring of eos6tg8

**Comment by GemmaTuron** [comment]
@samuelmaina 
Please also add to the list working on model eos7pw8 to get the .tar.gz file directly and install syba from there to avoid package import issues
I won't assign new tasks until we can close the work on `eos7pw8` and `eos6tg8`

**Comment by samuelmaina** [comment]
- [ ]  Work on the  eos6tg8
- [ ] Test model eos78ao -  https://github.com/ersilia-os/eos78ao/issues/7

**Comment by GemmaTuron** [comment]
@samuelmaina 

Thanks for the update, please let me know when the eos6tg8 is completed. Also, the eos7pw8 needs work as we discussed in the meeting yesterday (to get the syba package from conda directly).
Once these two and model testing are completed I'll assign new tasks

**Comment by samuelmaina** [comment]
Hello @GemmaTuron , 
Sorry for the late reply. Working on the issues.

**Comment by samuelmaina** [comment]
###  Friday,  15th July, 2023

- [ ]  Worked on  completing eos6tg8 working by converting the smiles data into dataframes. I tried to change the model input from dataframe to smiles  with no avail but the model had taken a lot of time , I decided to convert the read smiles into a dataframe.
- [ ]  Retested  eos46ev - https://github.com/ersilia-os/eos46ev/issues/5#issuecomment-1636145843
- [ ]  Incorporated the local changes for the ersilia fetch worflow
- [ ]  Retested https://github.com/ersilia-os/eos78ao/issues/7#issuecomment-1635750868

**Comment by samuelmaina** [comment]
- [ ] Continue with the incorporation of automation file creation
- [ ]  Handle the torch error in  eos6tg8 and submit a PR  - https://github.com/ersilia-os/eos6tg8/pull/3
- [ ]  Write the config  file suggestion issue in ersilia.

**Comment by samuelmaina** [comment]
## Monday, 17th July 2023
- [ ] Redo eos6tg8 and succcessfully have the PR merged- https://github.com/ersilia-os/eos6tg8/pull/5
- [ ] Refactor and Dockerzie eos1bba - https://github.com/ersilia-os/eos1bba/pull/20
- [ ] Made corrections and retested some models, ie. oes7pw8, eos1amr

**Comment by samuelmaina** [comment]
Hi @GemmaTuron ,
I don't have a lot of work today except for where I am waiting clarification or stuck. Maybe you can assign me some tasks.

**Comment by samuelmaina** [comment]
## 19th July, 2024

- [ ] Refactor eos6o0z  https://github.com/ersilia-os/eos6o0z/issues/1?notification_referrer_id=NT_kwDOA1VX87M3MDI3OTYyODQ2OjU1OTI0NzIz#event-9846990150
- [ ] Refactor  eos3xip  https://github.com/ersilia-os/eos3xip/issues/1?notification_referrer_id=NT_kwDOA1VX87M3MDI3OTYwMDA3OjU1OTI0NzIz#event-9846989641
- [ ] Solve  eos1amr issue  the https://github.com/ersilia-os/eos1amr/issues/4

**Comment by samuelmaina** [comment]
### Friday, 21st July, 2023
- [ ]  Tested model https://github.com/ersilia-os/eos1579/issues/3#issuecomment-1645309832
- [ ]  Invested the `float does not have property len` error for other models  single float models. For those I have tested, I came to realize that the behavior of the first `run` and consecutive `run`s  failure is common. I am also getting `index out of range error` when doing boundary eml datasize of 100, 200 or 300
- [ ]  Held a 1:1 meeting with my mentor. Discussed about the  number two above in depth and how to go.

**Comment by GemmaTuron** [comment]
Hi @samuelmaina,

Can you clarify what models you refer to in point 2?

I've added a couple of models to work on while we solve the issue with the workflows

**Comment by samuelmaina** [comment]
> Hi @samuelmaina,
> 
> Can you clarify what models you refer to in point 2?
> 
> I've added a couple of models to work on while we solve the issue with the workflows

 I have found a solution for float error  at the issue at https://github.com/ersilia-os/eos7pw8/issues/9#issuecomment-1649941152, I was testing `eos8lok ` for different batches but it was giving me error on the boundary numbers(100, 200).

**Comment by samuelmaina** [comment]
## Tuesday, 25th July, 2023

- [ ] Searched for a solution and test the solution https://github.com/ersilia-os/eos7pw8/issues/9#issuecomment-1649941152 for  eos1amr, eos9tyg , eos81ok.
- [ ] Attended the lab meeting and explained the above issue to the other members.
- [ ] Started refactoring `eos54c7 `- https://github.com/ersilia-os/eos54c7/issues/1#event-9908076561

Today spend most of my time on some personal issues I was handling but I was able to debug some single float models and  also attend the weekly update meeting. I also started with the refactoring of the model `eos54c7`, Able to put the model code into  a single function. Will work on the integration tomorrow.

**Comment by samuelmaina** [comment]
### Wednesday, 26th July, 2023
- [ ] Finish refactoring eos54c7 
- [ ] Finish refactoring  eos2a9n
   Both above models are supposed  to  scrap 100 molecules from a third part  website.The model were producing an output for the eml_data_set by I came to realize that the  output was not accurate.  The main.py was adding a constatnt to the Molecular Weight  of each molecule but the output is supposed to be  100 molecules closest  and their closeness values to the current smiles according to some specifc dataset.  I have tried to reach to the server API where the models are supposed to scrap but the server is dead. The publication has some data  and some training code , which I think can be used to train a  new querying system or maybe we can search for alternative API or tool to fetch from.
- [ ]  Meeting with my mentor.
 Discussed about the error resolution. for the eos1amr . Also planned on trial of the automation in real models to see their behavior.

**Comment by samuelmaina** [comment]
### Wednesday, 26th July, 2023
- [ ] Finish refactoring eos54c7 
- [ ] Finish refactoring  eos2a9n
   Both above models are supposed  to  scrap 100 molecules from a third part  website.The model were producing an output for the eml_data_set by I came to realize that the  output was not accurate.  The main.py was adding a constatnt to the Molecular Weight  of each molecule but the output is supposed to be  100 molecules closest  and their closeness values to the current smiles according to some specifc dataset.  I have tried to reach to the server API where the models are supposed to scrap but the server is dead. The publication has some data  and some training code , which I think can be used to train a  new querying system or maybe we can search for alternative API or tool to fetch from.
- [x]  Meeting with my mentor.
 Discussed about the error resolution. for the eos1amr . Also planned on trial of the automation in real models to see their behavior.

**Comment by samuelmaina** [comment]
### Thursday 27th July, 2023
- [ ] Worked with eos54c7 and eos2a9n to see if the request would go through.
- [ ] Attended the lab meeting.

**Comment by samuelmaina** [comment]
## Friday , 28th July, 2023

- [ ] continue working with file automation.Looking for the exact location for the files.
- [ ] Promote the config.yaml in slack.

**Comment by samuelmaina** [comment]
### Monday, 31st July, 2023

- [ ]  Continue working on the file automation. Created classes for the new models
- [ ] Refactor eos5qfo- Spent time looking on the `segment fault error `rom the main.py.

**Comment by GemmaTuron** [comment]
Hi @samuelmaina 
Let me know what you are working on today, there is a few model tests to be completed.
If you are not able to go around eos5qfo segmentation fault, let me know and we will add someone else to help you

**Comment by samuelmaina** [comment]
### Wednesday, 2nd Aug, 2023
- [ ]  Refactored eos2a9n and made PR - https://github.com/ersilia-os/eos2a9n/pull/2
- [ ]  Tested  eos85a3 https://github.com/ersilia-os/eos85a3/issues/4#issuecomment-1661088074
- Today I have been trying to resolve the segment error and trying to find a way in which models eos5qfo ,[eos2a9n](https://github.com/ersilia-os/eos2a9n) and https://github.com/ersilia-os/eos54c7/ could be be done away with.With miquel's help I think I will be able to finish them tomorrow quickly.
- I have some 2 models which I was to test. I though the  bugs would take shorter time. I will start with them tomorrow so that the models can be closed.

**Comment by samuelmaina** [comment]
### Friday August 4th,
- [ ] Tested eos715v
- [ ]  Rerun falling models eos2a9n to solve their issue

**Comment by samuelmaina** [comment]
### Monday 7th August, 2023

- [ ]  Refactored the https://github.com/ersilia-os/eos5qfo- Have been  stuggling with the `Allocation of 38535168 exceeds 10% of system memory`. I have tried to reduce the model batches and the size of the batches but  the error passists or other errors are thrown.
- [ ]  Solved the beautifulsoup input error of `no module bs4   found` even though `Beautifulsoup4` is in the Dockerfile.
- [ ]  Retested  eos85a3 https://github.com/ersilia-os/eos85a3/issues/4#issuecomment-1668597276
- [ ]  For model eos54c7 https://github.com/ersilia-os/eos54c7/issues/1 , The is not producing any result which is making the  fetch to fail. @GemmaTuron , I can't make a PR due to the fetch failure. You can use my fork to check the model state at https://github.com/samuelmaina/eos54c7

**Comment by samuelmaina** [comment]
### Tuesday, 8th August, 2023

- [ ] Test model eos1vms - https://github.com/ersilia-os/eos1vms/issues/2
- [ ] Solve for the `Allocation of 38535168 exceeds 10% of system memory' for  eos5qfo
- [ ]  Work together with @GemmaTuron , to make sure the models [eos2a9n](https://github.com/ersilia-os/eos2a9n) and [eos54c7](https://github.com/ersilia-os/eos54c7) are producing outputs.

**Comment by samuelmaina** [comment]
### Wednesday, 9th August, 2023

- [ ]  Resolve the parsing issue with `eos5qfo`
- [ ]  Incorporate model https://github.com/ersilia-os/eos7emx
- [ ]  Resolve minor arising from testing for the models that I previously incorporated.

**Comment by GemmaTuron** [comment]
Hi @samuelmaina 

I have added comments to the eos5fqo this morning so you could continue the work, and also asked a few questions about the model you are incorporating,

**Comment by samuelmaina** [comment]
#### Thursday,  10th August,
- [ ] Incorporate model eos7emx https://github.com/ersilia-os/eos7emx 
- [ ]  Reverse the dockerfile to the py 3.6 version for eos5qfo
- [ ]  Attend the lab meeting

**Comment by samuelmaina** [comment]
#### Thursday,  10th August,
- [ ] Full incorporation of the  model eos7emx https://github.com/ersilia-os/eos7emx  with expansion of the output 
as from this comment https://github.com/ersilia-os/eos7emx/pull/1#issuecomment-1673201211 .
- [ ] Make  a PR on eos5qfo. 
- [ ]  Test the test module by pittmanriley  and febielin

**Comment by GemmaTuron** [comment]
Hi @samuelmaina 
I have left comments both for model eos7emx and eos5qfo, let's try to close on this tasks!

**Comment by samuelmaina** [comment]
Thanks very much  @GemmaTuron , 
I will have a look at them.

**Comment by samuelmaina** [comment]
### Monday 14th, August 

- [x] Solve the `example`  not able to produce samples for other input shapes unless for single compounds. Documented the issue and the solution at https://github.com/ersilia-os/eos5qfo/pull/2#issuecomment-1678457550
- [] Increment the length of the sample for https://github.com/ersilia-os/eos7emx

**Comment by GemmaTuron** [comment]
Hi @samuelmaina,

Aside from the Documentation improvements, there are still a few issues open that are assigned to you, please make sure to tackle them and let me know when PR's are ready to be reviewed:

- Eos5qfo: clarify the input used in the model and make sure it works with either a single list or several lists
- SmallWorld: please revise my comments and ammend the model, it is missing a big part of the original code
- eos2a9n: make sure it calculates all molecules
- eos54c7: make sure the url is correct and the model works
